lib/open_project/translations/models/locales_updater.rb

```diff
@@ -0,0 +1,157 @@
+# todo those ought to be in the gemspec
+require 'tempfile'
+require 'crowdin-api'
+require 'zip'
+require 'pathname'
+require 'fileutils'
+
+require_relative '../helpers/tmp_directory'
+require_relative './git_repository'
+require_relative './i18n_provider'
+require_relative './locales_updater_configuration'
+
+ENGLISH_TRANLATION_FILE = 'en.yml'
+ENGLISH_JS_TRANLATION_FILE = 'js-en.yml'
+ACCEPTANCE_LEVEL = ENV['ACCEPTANCE_LEVEL'] || 100
+
+class LocalesUpdater
+  extend TmpDirectory
+
+  def self.update_all_locales_of_all_repos(debug: true)
+    repos_to_update = plugins_with_locales
+
+    within_tmp_directory(delete_if_exists: true, debug: debug) do
+      repos_to_update.each do |name, specifics|
+        # todo each branch
+        create_i18n_handle(specifics)
+
+        within_tmp_directory(path: File.join(FileUtils.pwd, name), debug: debug) do
+          uri = specifics[:uri]
+          git_repo = GitRepository.new(uri, FileUtils.pwd)
+          git_repo.clone
+
+          git_repo.checkout(branch)
+          # todo or should we merge this branch into the next ('push vs pull')
+          git_repo.merge(previous_branch, strategy: :ours) if previous_branch
+          git_repo.within_repo do
+            # todo rescue should be in provider class
+            begin
+              upload_english
+
+              request_build
+              download_and_replace_locales
+            rescue Crowdin::API::Errors::Error => e
+              puts "Error during update of #{name}: #{e.message}"
+              exit
+            end
+          end
+          git_repo.add('config/locales')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.